File: appJava/src/main/java/game/SocketClient.java
package game;

import java.io.IOException;
import java.net.Socket;
import org.json.simple.JSONObject;

import java.io.DataOutputStream;

public class SocketClient {
    Socket client;
    DataOutputStream dOut;

    public SocketClient(String ip ,int port) throws IOException {
        client = new Socket(ip,port);
        dOut = new DataOutputStream(client.getOutputStream());
        // Message mess = new Message(1, 1, ePiece.PAPER);
        // sendMessage(mess);
    }

    public void sendMessage(Message message) throws IOException
    {
        JSONObject json = message.createJsonMessage();
        dOut.writeUTF(json.toJSONString());
        dOut.flush();
    }

    protected void finalize() throws IOException
    {
        dOut.close();
        client.close();
    }
}

File: appPython/SocketClient.py
from Message import Message
from ePiece import ePiece
import socket

class SocketClient():
    

    def __init__(self, ip, port):
        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        server_addr = (ip, port)
        self.sock.connect(server_addr)

    def sendMessage(self, message): 
        jsonObj = message.createJsonMessage();
        self.sock.send(jsonObj.encode("utf-8"))
    

    def __del__(self): 
        self.sock.close()
